### PR TITLE
Feature search by cuisine

### PR DESCRIPTION
**Pull request by sagar-13**
**Ignore this PR until I merge with dev after #41  is merged.** 

- New /search endpoint 
- Returns all chefs or all meals depending on required parameter _searchType_
- Can also retrieve chefs/meals with specific cuisineType/cuisineSpecialty (supplied in the _cuisine_ parameter)

### CODE REVIEW

File: server/routes/search.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const chefsController = require("../controllers/chefsController");
+const mealsController = require("../controllers/mealsController");
+
+
+router.get("/", async (req, res) => {
+
+    if (! ["chefs", "meals"].includes(req.query.searchType)) {
+        res.status(400).json({ errors: ["searchType should be one of chefs or meals!"] });
+        return;
+    }
```
Comment (sina-jamshidi) [inline server/routes/search.js]: Nice, that works

File: server/routes/search.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if (req.query.searchType === "chefs"){
+        const query = req.query.cuisine ?  {cuisineSpecialty: {$all: [req.query.cuisine]}} : {}
+        chefs = await chefsController.findAllChef(query);
+        res.json(chefs);
+    }
+    if (req.query.searchType === "meals"){
+        const query = req.query.cuisine ?  {cuisineType: {$all: [req.query.cuisine]}} : {}
+        meals = await mealsController.findAllMeals(query);
+        res.json(meals);
+    }
```
Comment (sina-jamshidi) [inline server/routes/search.js]: Nice. One extension to work on here is to potentially accept multiple cuisines (perhaps comma separated?)

File: server/routes/search.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    return;
+    
+    
+    })
+
```
Comment (gary-bains) [inline server/routes/search.js]: so much white space

**Comment by sagar-13** [comment]
So, I changed it to correspond with the google doc specification. 

if searchType is chefs, you get a list of meals. Each meal will have an additional chef property (not just the id, whole chef). 

If searchType is meals, you get an object like:
 {
"meals":  list of meals,
 "chefs": {"chef._id": full chef object }
}

File: server/models/meal.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const mealSchema = new Schema({
     requirements: String,
     createdAt: { type: Date, default: Date.now },
     updatedAt: Date,
-    chefId: { type: Schema.Types.ObjectId, ref: "Chefs" },
+    chefId: { type: Schema.Types.ObjectId, ref: "Chef" },
```
Comment (gary-bains) [inline server/models/meal.js:15]: shouldn't it be `Chefs`

**Review by gary-bains — APPROVED** [review]
looks good

File: server/models/meal.js (hunk 4 above)
Comment (sagar-13) [reply server/models/meal.js:15]: I don't think so. I only changed it for consistency with the User and Meal models.

**Pull request merged**